src/DotNetWorker.Grpc/GrpcWorker.cs

```diff
@@ -210,24 +210,20 @@ internal static WorkerInitResponse WorkerInitRequestHandler(WorkerInitRequest re
 
             response.WorkerMetadata.CustomProperties.Add("Worker.Grpc.Version", typeof(GrpcWorker).Assembly.GetName().Version?.ToString());
 
-            response.Capabilities.Add("RpcHttpBodyOnly", bool.TrueString);
-            response.Capabilities.Add("RawHttpBodyBytes", bool.TrueString);
-            response.Capabilities.Add("RpcHttpTriggerMetadataRemoved", bool.TrueString);
-            response.Capabilities.Add("UseNullableValueDictionaryForHttp", bool.TrueString);
-            response.Capabilities.Add("TypedDataCollection", bool.TrueString);
-            response.Capabilities.Add("WorkerStatus", bool.TrueString);
-            response.Capabilities.Add("HandlesWorkerTerminateMessage", bool.TrueString);
-            response.Capabilities.Add("HandlesInvocationCancelMessage", bool.TrueString);
-
-            if (workerOptions.EnableUserCodeException)
+            // Add additional capabilities defined by WorkerOptions
+            foreach ((string key, string value) in workerOptions.Capabilities)
             {
-                response.Capabilities.Add("EnableUserCodeException", bool.TrueString);
-            }
-            if (workerOptions.IncludeEmptyEntriesInMessagePayload)
-            {
-                response.Capabilities.Add("IncludeEmptyEntriesInMessagePayload", bool.TrueString);
+                response.Capabilities[key] = value;
             }
 
+            // Add required capabilities; these cannot be modified and will override anything from WorkerOptions
```

Comment:
Doesn't really make a difference, but would expect the required items to be added before the additional capabilities

Reply:
I wanted to make sure that the required ones "win" -- so even if you set one of these to false, it ends up being true.